internal/ascpverifier/structured_data.go

```
@@ -0,0 +1,55 @@
+	case <-ctx.Done():
+		return EngineResult{}, ctx.Err()
+	default:
+	}
+	if input.Spec.Class != ClassStructuredData || input.Spec.Tolerance != "0" || input.Spec.ReferenceSource != "captured-delivery" {
+		return EngineResult{}, fmt.Errorf("%w: structured-data engine requires exact self-contained evidence", ErrInvalidEngineResult)
+	}
+	decoder := json.NewDecoder(bytes.NewReader(input.Delivery.Content))
+	decoder.UseNumber()
+	var document map[string]any
+	if err := decoder.Decode(&document); err != nil || document == nil {
+		return EngineResult{Verdict: VerdictFail, Code: "invalid-json-object"}, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 29) but it returns nil

(nilerr)